### DISCUSSION TITLE
unsure of the correct way to go about adding a user to an event

### DISCUSSION THREAD

**Discussion by colinpcurtis** [prisma/prisma#10914] [Q&A] [answered]
I have a model that looks like this

```prisma 
model User {
    id          String      @id @default(uuid())
    email       String      @unique
    name        String
    eventsMade  Event[]     @relation("createdBy")
    profile     Profile?
    attending   Event[]
}

model Profile {
    id          String      @id @default(uuid())
    user        User        @relation(fields: [userId], references: [id])
    userId      String 
    bio         String
}
model Event {
    id          String      @id @default(uuid())
    title       String
    createdBy   User        @relation(name: "createdBy", fields: [userId], references: [id])
    userId      String
    filePath    String 
    eventDate   DateTime
    bio         String?
    attending   User[]      @relation(references: [id])
}
```

and right now I'm trying to implement a feature that will add a user to an event, basically so that the user knows they're attending the event.  I've tried a lot of iterations to try to get this work, and sometimes I can get it to work but I always get stuck on this and the deleting a user from the event. 

Right now my current code looks like this 
```js
        const user = await prisma.user.findUnique({
            where: {
                id: req.body.userId
            }
        })
        const event = await prisma.event.update({
            where: {
                id: req.body.eventId
            },
            data: {
                attending: {
                    push: user
                } 
            }
        })
```

and I'm getting an error that looks like

```
Unknown arg `attending` in data.attending for type EventUncheckedUpdateInput. Available args:
api: type EventUncheckedUpdateInput {
api:   id?: String | StringFieldUpdateOperationsInput
api:   title?: String | StringFieldUpdateOperationsInput
api:   userId?: String | StringFieldUpdateOperationsInput
api:   filePath?: String | StringFieldUpdateOperationsInput
api:   eventDate?: DateTime | DateTimeFieldUpdateOperationsInput
api:   bio?: String | NullableStringFieldUpdateOperationsInput | Null
api: }
```

I'm kind of stuck as to how to correctly implement this feature so that I can add users to an event.  I assume this is a decently efficient way to model such a relationship, but if there's a better way to do something like this, I'd be very glad to know as well.

**Comment by ajmnz** [comment, marked as answer]
From [the docs](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#push):
> Use push to add one value **to a scalar list field**.

I believe relations don't apply as scalar list fields.


In this case, you need to [`connect`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#connect) the `attending` field to a user `id`.

```ts
const event = await prisma.event.update({
  where: {
    id: req.body.eventId,
  },
  data: {
    attending: {
      connect: {
        id: user.id,
      },
    },
  },
});
```

To revert it, you just have to use [`disconnect`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#disconnect) instead of `connect`.

```ts
data: {
  attending: {
    disconnect: {
      id: user.id,
    },
  },
},
```

Let me know if that works!

**Reply by colinpcurtis** [reply]
I decided to make a different design decision that ended up being a much cleaner way to implement the feature with the following schema

```prisma
model User {
    id          String      @id @default(uuid())
    email       String      @unique
    name        String
    eventsMade  Event[]     @relation("createdBy")
    profile     Profile?
    attending   Attending[]
}

model Profile {
    id          String      @id @default(uuid())
    user        User        @relation(fields: [userId], references: [id])
    userId      String 
    bio         String
}

model Event {
    id          String      @id @default(uuid())
    title       String
    createdBy   User        @relation(name: "createdBy", fields: [userId], references: [id])
    userId      String
    filePath    String 
    eventDate   DateTime
    bio         String?
    attending   Attending[]
}

model Attending {
  id            String      @id @default(uuid())
  event         Event       @relation(fields: [eventId], references: [id])
  user          User        @relation(fields: [userId], references: [id])
  eventId       String
  userId        String
}
```

before I saw that you had commented.  

In particular, I believe that this design choice makes it much easier to see the relations among who is coming to events, since
you just have to make a simple `prisma.attending.create` or `delete` call to modify who is coming to an event instead of having to deal with complicated lists.  

Thanks for your help and hopefully this can help someone in the future.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
